db/collection.go

```diff
@@ -170,21 +169,17 @@ func (c *Collection) Update(model Model) error {
 // package, model must be settable via reflect. Typically, this means you should
 // pass in a pointer.
 func (c *Collection) FindByID(id []byte, model Model) error {
-	pk := c.primaryKeyForID(id)
-	return c.findByKey(pk, model)
-}
-
-func (c *Collection) findByKey(key []byte, model Model) error {
 	if err := c.checkModelType(model); err != nil {
 		return err
 	}
-	data, err := c.db.ldb.Get(key, nil)
+	pk := c.primaryKeyForID(id)
+	data, err := c.db.ldb.Get(pk, nil)
 	if err != nil {
+		if err == leveldb.ErrNotFound {
+			return NotFoundError{ID: id}
+		}
 		return err
 	}
-	if data == nil {
-		return errors.New("model not found")
-	}
```

Comment:
Why is this conditional no longer needed?

Reply:
It was probably never needed. If the key isn't found in the database it returns a `leveldb.ErrNotFound` instead of `nil`.

If you are using the API exposed in the db package, data cannot be nil. Technically, someone could manually add a key with a nil value, but there are so many other things you could break by manually setting keys that I think it would be too hard to consider.